store/store.go

```diff
@@ -2,108 +2,113 @@
 package store
 
 import (
-	"context"
-	"time"
+	"errors"
+	"os"
+	"sync"
 
-	"github.com/gluster/glusterd2/gdctx"
+	"github.com/gluster/glusterd2/pkg/elasticetcd"
 
-	log "github.com/Sirupsen/logrus"
 	"github.com/coreos/etcd/clientv3"
 	"github.com/coreos/etcd/clientv3/concurrency"
-	"github.com/pborman/uuid"
-	config "github.com/spf13/viper"
 )
 
 const (
 	// GlusterPrefix prefixes all paths in the store
-	GlusterPrefix     = "gluster/"
-	sessionTTL        = 30 // used for etcd mutexes and liveness key
-	livenessKeyPrefix = GlusterPrefix + "alive/"
+	GlusterPrefix = "gluster/"
+	sessionTTL    = 30 // used for etcd mutexes and liveness key
 )
 
-// Store variable can be imported by packages which need access to the store
-var Store *GDStore
+var (
+	// Store is the default GDStore that must to be used by the packages in GD2
+	Store *GDStore
```

Comment:
I'm trying to find a way to correctly guard this.
This current method with the mutex is half baked, not all access is guarded. In this method I'd have to wrap the all the methods of the etcd client, with a new function that uses this lock. This doesn't feel nice to me.